### PR TITLE
tikv-ctl: fix tikv-ctl's output is incomplete on calling process::exit

### LINKED ISSUE #11210: tikv-ctl's output is incomplete, maybe slog-async is not flushed when tikv-ctl process exits
I am using the master branch of tikv-server which build from source code. And running the following tikv-ctl command gives incomplete outputs:
```bash
$ ./tikv-ctl --data-dir=/path/to/tikv/data/dir size
[2021/11/02 20:51:48.157 +08:00] [INFO] [mod.rs:118] ["encryption: none of key dictionary and file dictionary are found."]
[2021/11/02 20:51:48.157 +08:00] [INFO] [mod.rs:479] ["encryption is disabled."]
[2021/11/02 20:51:48.160 +08:00] [WARN] [config.rs:338] ["Memory allocator nodump is not supported, continue with default allocator"]
[2021/11/02 20:51:48.164 +08:00] [WARN] [config.rs:587] ["compaction guard is disabled due to region info provider not available"]
[2021/11/02 20:51:48.164 +08:00] [WARN] [config.rs:682] ["compaction guard is disabled due to region info provider not available"]
```
as shown above, some logs not printed, and if add `std::thread::sleep(std::time::Duration::from_secs(1));` before 
https://github.com/tikv/tikv/blob/8a16a5b63402096850509855961e7f1be5705509/cmd/tikv-ctl/src/main.rs#L92
the output is complete now:
```bash
$ ./tikv-ctl --data-dir=/path/to/tikv/data/dir size
[2021/11/02 21:11:29.503 +08:00] [INFO] [mod.rs:118] ["encryption: none of key dictionary and file dictionary are found."]
[2021/11/02 21:11:29.503 +08:00] [INFO] [mod.rs:479] ["encryption is disabled."]
[2021/11/02 21:11:29.506 +08:00] [WARN] [config.rs:317] ["Memory allocator nodump is not supported, continue with default allocator"]
[2021/11/02 21:11:29.510 +08:00] [WARN] [config.rs:587] ["compaction guard is disabled due to region info provider not available"]
[2021/11/02 21:11:29.510 +08:00] [WARN] [config.rs:682] ["compaction guard is disabled due to region info provider not available"]
[2021/11/02 21:11:29.514 +08:00] [ERROR] [main.rs:82] ["error while open kvdb: Storage Engine IO error: While lock file: /path/to/tikv/data/dir/db/LOCK: Resource temporarily unavailable"]
[2021/11/02 21:11:29.514 +08:00] [ERROR] [main.rs:85] ["LOCK file conflict indicates TiKV process is running. Do NOT delete the LOCK file and force the command to run. Doing so could cause data corruption."]
```

and if I set `use_async`
https://github.com/tikv/tikv/blob/8a16a5b63402096850509855961e7f1be5705509/components/tikv_util/src/logger/mod.rs#L78
to always false , the output is complete ditto.

So, maybe tikv-ctl needs to use sync' mode slog to different from tikv-server which is performance sensitive.

### PR DESCRIPTION
**Pull request by yaozongyou**
Signed-off-by: Yao Zongyou <yaozongyou@vip.qq.com>

### What problem does this PR solve?

Issue Number: close #11210

Problem Summary: the output of tikv-ctl is incomplete on calling process::exit.

### What is changed and how it works?

What's Changed: force async logger to flush by dropping its guard.

### Release note
```release-note
None
```

### CODE REVIEW

**Review by Connor1996 — COMMENTED** [review]
Let's use `println` directly

File: cmd/tikv-ctl/src/main.rs [hunk 1]
```diff
@@ -89,6 +89,10 @@ fn handle_engine_error(err: EngineError) -> ! {
             );
         }
     }
+
+    // force async logger to flush by dropping its guard.
+    *tikv_util::logger::ASYNC_LOGGER_GUARD.lock().unwrap() = None;
+
     process::exit(-1);
```
Comment (Connor1996) [inline cmd/tikv-ctl/src/main.rs:96]: `process::exit(-1)` is called in elsewhere as well, please extract a function.

Comment (yaozongyou) [reply cmd/tikv-ctl/src/main.rs:96]: sorry, I try to rebase my pull request on master, and then I can not reproduce the issue at this moment.
I will dig into the source code and see the reasons behind this and I will keep you updated with the progress.

Comment (yaozongyou) [reply cmd/tikv-ctl/src/main.rs]: my bad, this issue can still be reproduced by executing many and many times or by adding more error! print sentences before process::exit(-1).

**Comment by yaozongyou** [comment]
ping @Connor1996

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by tabokie — APPROVED** [review]

**Pull request merged**